src/com/oltpbenchmark/benchmarks/tpcc/TPCCLoader.java

```diff
@@ -148,15 +149,25 @@ protected void transRollback(Connection conn) {
     }
   }
 
-  protected void transCommit(Connection conn) {
-    try {
-      conn.commit();
-    } catch (SQLException se) {
-      LOG.debug(se.getMessage());
-      transRollback(conn);
+  private void performInserts(List<PreparedStatement> stmts, Connection conn) throws Exception {
+    int attempts = workConf.getMaxLoaderRetries() + 1;
+    for (int i = 0; i < attempts; ++i) {
+      try {
+         for (PreparedStatement stmt: stmts) {
+            stmt.executeBatch();
+            stmt.clearBatch();
+         }
+          conn.commit();
+          return;
+      } catch (SQLException ex) {
+        transRollback(conn);
+        if (i == attempts - 1) {
+          LOG.error("Failed to load data for TPCC", ex);
+          throw ex;
+        }
```

Comment:
lets at least LOG.warning even if its not the last attempt

Reply:
Done.